lib/form-persister.js

```diff
@@ -0,0 +1,38 @@
+export default class FormPersister {
+  constructor(formId = null, throttlePeriod = 1000) {
+    this.formId = formId;
+    this.throttlePeriod = throttlePeriod;
+    this.canSave = true;
+    this.values = null;
+  }
+
+  setFormId(formId) {
+    this.formId = formId;
+  }
+
+  saveValues(values) {
+    this.values = values;
+    if (this.canSave && this.formId) {
+      this.canSave = false;
+      window.localStorage.setItem(this.formId, JSON.stringify(this.values));
+      setTimeout(() => {
+        this.canSave = true;
+        window.localStorage.setItem(this.formId, JSON.stringify(this.values));
+      }, this.throttlePeriod);
+    }
+  }
+
+  loadValues() {
+    if (this.formId) {
+      const itemFromStorage = window.localStorage.getItem(this.formId);
+      return JSON.parse(itemFromStorage);
+    }
+    return null;
+  }
+
+  clearValues() {
```

Comment:
Great that you introduced this, it wasn't in the spec but we should definitely clear old values after we're done with them!